src/Exceptions/YourMembershipException.php

```diff
@@ -0,0 +1,13 @@
+<?php
+namespace P2A\YourMembership\Exceptions;
+
+/**
+ * Generic YourMembershipException
+ *
+ * Used typically for errors dealing with XML Parsing Issues
+ *
+ */
+class YourMembershipException extends \Exception
```

Comment:
Why the superclass? The difference between YourMembershipException and YourMembershipApiException isn't clear to me. From the perspective of a client of your package, they are all sort of the same type of thing, right? Maybe there is some purpose internal to this package.